src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -73,356 +72,310 @@ const defaultProps = {
-        if (this.isEditing) {
-            return this.props.translate('iou.amount');
-        }
-        const title = {
-            [CONST.IOU.MONEY_REQUEST_TYPE.REQUEST]: this.props.translate('iou.requestMoney'),
-            [CONST.IOU.MONEY_REQUEST_TYPE.SEND]: this.props.translate('iou.sendMoney'),
-            [CONST.IOU.MONEY_REQUEST_TYPE.SPLIT]: this.props.translate('iou.splitBill'),
-        };
-        return title[this.iouType];
-    }
+    const title = {
+        [CONST.IOU.MONEY_REQUEST_TYPE.REQUEST]: translate('iou.requestMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SEND]: translate('iou.sendMoney'),
+        [CONST.IOU.MONEY_REQUEST_TYPE.SPLIT]: translate('iou.splitBill'),
+    };
+    const titleForStep = isEditing.current ? translate('iou.amount') : title[iouType.current];
 
     /**
      * Check and dismiss modal
      */
-    dismissModalIfNecessary() {
-        if (!ReportUtils.shouldHideComposer(this.props.report, this.props.errors)) {
+    useEffect(() => {
+        if (!ReportUtils.shouldHideComposer(props.report, props.errors)) {
             return;
         }
-        Navigation.dismissModal(this.reportID);
-    }
+        Navigation.dismissModal(reportID.current);
+    }, [props.errors, props.report]);
 
     /**
      * Focus text input
      */
-    focusTextInput() {
+    const focusTextInput = () => {
         // Component may not initialized due to navigation transitions
         // Wait until interactions are complete before trying to focus
         InteractionManager.runAfterInteractions(() => {
             // Focus text input
-            if (!this.textInput) {
+            if (!textInput.current) {
                 return;
             }
 
-            this.textInput.focus();
+            textInput.current.focus();
         });
-    }
+    };
 
-    /**
-     * @param {String} amount
-     * @returns {Number}
-     */
-    calculateAmountLength(amount) {
-        const leadingZeroes = amount.match(/^0+/);
-        const leadingZeroesLength = lodashGet(leadingZeroes, '[0].length', 0);
-        const absAmount = parseFloat((this.stripCommaFromAmount(amount) * 100).toFixed(2)).toString();
-
-        // The following logic will prevent users from pasting an amount that is excessively long in length,
-        // which would result in the 'absAmount' value being expressed in scientific notation or becoming infinity.
-        if (/\D/.test(absAmount)) {
-            return CONST.IOU.AMOUNT_MAX_LENGTH + 1;
+    useEffect(() => {
+        if (isEditing.current) {
+            if (prevMoneyRequestId.current !== props.iou.id) {
+                // The ID is cleared on completing a request. In that case, we will do nothing.
+                if (props.iou.id) {
+                    Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
+                }
+                return;
+            }
+            const moneyRequestId = `${iouType.current}${reportID.current}`;
```

Comment:
I believe to comply with our naming guides this should be uppercase ID. Could you update this and other similar instances please?
```suggestion
            const moneyRequestID = `${iouType.current}${reportID.current}`;
```